lua/entities/gmod_wire_expression2/core/egpobjects.lua

```diff
@@ -0,0 +1,493 @@
+	return (this.w and this.h and { this.w, this.h }) or { -1, -1 }
+end
+
+e2function number egpobject:egpSizeNum()
+	if not this then return self:throw("Invalid EGP Object", -1) end
+	return this.size or -1
+end
+
+e2function vector4 egpobject:egpColor4()
+	if not this then return self:throw("Invalid EGP Object", { -1, -1, -1, -1 }) end
+	return (this.r and this.g and this.b and this.a and { this.r, this.g, this.b, this.a }) or { -1, -1, -1, -1 }
+end
+
+e2function vector egpobject:egpColor()
+	if not this then return self:throw("Invalid EGP Object", -1) end
+	return (this.r and this.g and this.b and Vector(this.r, this.g, this.b)) or Vector(-1, -1, -1)
+end
+
+e2function number egpobject:egpAlpha()
+	if not this then return self:throw("Invalid EGP Object", -1) end
+	return this.a or -1
+end
+
+e2function number egpobject:egpAngle()
+	if not this then return self:throw("Invalid EGP Object", -1) end
+	return this.angle or -1
+end
+
+e2function string egpobject:egpMaterial()
+	if not this then return self:throw("Invalid EGP Object", "") end
+	return this.material or ""
+end
+
+e2function number egpobject:egpRadius()
+	if not this then return self:throw("Invalid EGP Object", -1) end
+	return this.radius or -1
+end
+
+__e2setcost(10)
+
+e2function array egpobject:egpVertices()
+	if (this.vertices) then
+		local ret = {}
+		for k, v in ipairs(this.vertices) do
+			ret[k] = { v.x, v.y }
+		end
+		return ret
+	elseif v.x and v.y and v.x2 and v.y2 and v.x3 and v.y3 then
+		return { {v.x, v.y}, {v.x2, v.y2 }, { v.x3, v.y3 } }
+	elseif v.x and v.y and v.x2 and v.y2 then
+		return { {v.x, v.y}, { v.x2, v.y2 } }
+	else
+		return {}
+	end
+end
+
+--------------------------------------------------------
+-- Object Type
+--------------------------------------------------------
+__e2setcost(4)
+
+e2function string egpobject:egpObjectType()
+	return EGP.Objects.Names_Inverted[this.ID] or ""
+end
+
+--------------------------------------------------------
+-- Additional Functions
+--------------------------------------------------------
+
+__e2setcost(15)
+
+e2function egpobject wirelink:egpCopy(number index, egpobject from)
+	if not EGP:IsAllowed(self, this) then return end
+	if from then
+		local copy = table.Copy(from)
+		copy.index = index
+		local bool, obj = EGP:CreateObject(this, from.ID, copy, self.player)
+		if bool then EGP:DoAction(this, self, "SendObject", obj) Update(self, this) return obj end
+	end
+end
+
+__e2setcost(10)
+
+e2function number egpobject:egpObjectContainsPoint(vector2 point)
+	return this and this:Contains(point[1], point[2]) and 1 or 0
+end
+
+__e2setcost(5)
+
+e2function egpobject wirelink:egpobject(number index)
+	if not EGP:IsAllowed(self, this) then return end
+	local _, _, obj = EGP:HasObject(this, index)
+	return obj
+end
+
+__e2setcost(2)
+
+e2function string egpobject:egpObjectType()
+	return this and EGP.Objects.Names_Inverted[this.ID] or ""
+end
+
+__e2setcost(1)
+
+e2function egpobject noegpobject()
+	return nil
```

Comment:
Don't think we should use nil for null values anymore. Just have a special null variant of the egp object